crates/uv-extract/src/stream.rs

```diff
@@ -75,7 +75,10 @@ pub async fn unzip<R: tokio::io::AsyncRead + Unpin>(
             let path = target.as_ref().join(path);
 
             if let Some(mode) = entry.unix_permissions() {
-                fs_err::set_permissions(&path, Permissions::from_mode(mode))?;
+                // Create a file with read and write permissions.
+                // Otherwise, databricks-0.2.dist-info/RECORD in databricks-0.2-py2.py3-none-any.whl has 000 permissions.
+                let permissions = Permissions::from_mode(mode | 0o660);
```

Comment:
Permission seems to lack an api to set read and write without manual bitflags math.

Reply:
This should take `umask` into account, see https://github.com/pypa/pip/blob/main/src/pip/_internal/operations/install/wheel.py#L652

Reply:
Thanks for the pip reference!

I rewrote it so that we're now never setting the permissions explicitly ourselves, but instead first create the file with the default options and then attach the executable bit if required.